src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -510,6 +511,21 @@ private AbstractFrame getFieldWithoutThis(DetailAST ast, int parentType) {
         return frame;
     }
 
+    /**
+     * Return whether ast is in a COMPACT_CTOR_DEF.
+     *
+     * @param ast The token to check
+     * @return true if ast is in a COMPACT_CTOR_DEF, false otherwise
+     */
+    private static boolean isInCompactConstructor(DetailAST ast) {
+        boolean isInCompactCtor = false;
+        if (ast.getParent() != null) {
+            isInCompactCtor = ast.getParent().getType() == TokenTypes.COMPACT_CTOR_DEF
+                                || isInCompactConstructor(ast.getParent());
```

Comment:
This can be easily replaced with a loop. How deep this recursion can be?

Reply:
You are right. I have replaced it with a loop.
The recursion can reasonably be up to 6-8 level deep, depending on how many nested loops, nested expressions there are.

Reply:
@pbludov The code has been fixed. Please review